streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -184,32 +217,21 @@ public void registerStore(final StateStore store, final StateRestoreCallback sta
 
         log.info("Restoring state for global store {}", store.name());
         final List<TopicPartition> topicPartitions = topicPartitionsForStore(store);
-        Map<TopicPartition, Long> highWatermarks = null;
 
-        int attempts = 0;
-        while (highWatermarks == null) {
-            try {
-                highWatermarks = globalConsumer.endOffsets(topicPartitions);
-            } catch (final TimeoutException retryableException) {
-                if (++attempts > retries) {
-                    log.error("Failed to get end offsets for topic partitions of global store {} after {} retry attempts. " +
-                        "You can increase the number of retries via configuration parameter `retries`.",
-                        store.name(),
-                        retries,
-                        retryableException);
-                    throw new StreamsException(String.format("Failed to get end offsets for topic partitions of global store %s after %d retry attempts. " +
-                            "You can increase the number of retries via configuration parameter `retries`.", store.name(), retries),
-                        retryableException);
-                }
-                log.debug("Failed to get end offsets for partitions {}, backing off for {} ms to retry (attempt {} of {})",
-                    topicPartitions,
-                    retryBackoffMs,
-                    attempts,
-                    retries,
-                    retryableException);
-                Utils.sleep(retryBackoffMs);
-            }
+        final Map<TopicPartition, Long> highWatermarks;
+        try {
+            highWatermarks = globalConsumer.endOffsets(topicPartitions);
+        } catch (final TimeoutException retryableException) {
+            log.debug(
+                "Failed to get end offsets for partitions {}. The broker may be transiently unavailable at the moment. Will retry.",
+                topicPartitions,
+                retryableException
```

Comment:
The varargs version of `debug` does _not_ take a cause at the end. This exception will not be logged. You have to use the version that only takes `(String, Exception)`.